synapse/models/dns.py

```diff
@@ -101,211 +107,76 @@ def getModelDefs(self):
-                    ('server:port', ('inet:ipv4', {}), {
-                        'doc': 'The port of the server that responded to the lookup.',
-                    }),
+
+                ('inet:dns:request', {}, (
+
+                    ('time', ('time', {}), {}),
+                    ('query', ('inet:dns:query', {}), {}),
+
+                    ('server', ('inet:server', {}), {}),
+
                     ('rcode', ('int', {}), {
-                        'doc': 'The DNS server response code.',
-                    }),
+                        'doc': 'The DNS server response code.'}),
+
                     ('exe', ('file:bytes', {}), {
-                        'doc': 'The file containing the code that attempted the DNS lookup.',
-                    }),
+                        'doc': 'The file containing the code that attempted the DNS lookup.'}),
+
                     ('proc', ('it:exec:proc', {}), {
-                        'doc': 'The process that attempted the DNS lookup.',
-                    }),
+                        'doc': 'The process that attempted the DNS lookup.'}),
+
                     ('host', ('it:host', {}), {
-                        'doc': 'The host that attempted the DNS lookup.',
-                    }),
-                    # one of the following should be set...
-                    ('a', ('inet:dns:a', {}), {
-                        'ro': True,
+                        'doc': 'The host that attempted the DNS lookup.'}),
+
+                )),
+
+                ('inet:dns:reply', {}, (
+
+                    ('ttl', ('int', {}), {}),
+                    ('request', ('inet:dns:request', {}), {}),
+
+                    ('a', ('inet:dns:a', {}), { 'ro': True,
                         'doc': 'The DNS A record returned by the lookup.',
                     }),
-                    ('a:fqdn', ('inet:fqdn', {}), {
-                        'ro': True,
-                        'doc': 'The domain queried for its A record.',
-                    }),
-                    ('a:ipv4', ('inet:ipv4', {}), {
-                        'ro': True,
-                        'doc': 'The IPv4 address returend in the A record.',
-                    }),
-                    ('ns', ('inet:dns:ns', {}), {
-                        'ro': True,
+                    ('ns', ('inet:dns:ns', {}), { 'ro': True,
                         'doc': 'The DNS NS record returned by the lookup.',
                     }),
-                    ('ns:zone', ('inet:fqdn', {}), {
-                        'ro': True,
-                        'doc': 'The domain queried for its NS record.',
-                    }),
-                    ('ns:ns', ('inet:fqdn', {}), {
-                        'ro': True,
-                        'doc': 'The domain returned in the NS record.',
-                    }),
-                    ('rev', ('inet:dns:rev', {}), {
-                        'ro': True,
+                    ('rev', ('inet:dns:rev', {}), { 'ro': True,
```

Comment:
Nitpick, but I would be in favor of changing 'rev' to 'ptr' not just here but for all relevant forms...it's the only DNS form that is misnamed vs. the real world.

Reply:
Yea, i can see that, but then the value we record would have to be 4.3.2.1.in-addr.arpa.  I'll take a look at tweaking that, and maybe making a secondary property on inet:ipv4 that we could then pivot to...

Reply:
Hadn't considered that - mostly it's a user thing, i'm always trying to pivot to inet:dns:ptr:ipv4 (for example) and then having a hurr derp moment.

Reply:
This issue probably requires more thought.  I'd prefer to table it for this delta so we can get it merged.  We can create a story about inet:dns:rev.